#! /bin/sh /usr/share/dpatch/dpatch-run
## disable-pie-mipsen.dpatch by Martin Pitt <mpitt@debian.org>
##
## DP: Disable PIE on mipsen, its binutils segfaults with it. (#548246)

@DPATCH@
diff -urNad cups-1.4.1~/config-scripts/cups-compiler.m4 cups-1.4.1/config-scripts/cups-compiler.m4
--- cups-1.4.1~/config-scripts/cups-compiler.m4	2009-05-17 02:13:47.000000000 +0200
+++ cups-1.4.1/config-scripts/cups-compiler.m4	2009-09-29 09:19:05.926504967 +0200
@@ -149,6 +149,7 @@
 	# The -pie option is available with some versions of GCC and adds
 	# randomization of addresses, which avoids another class of exploits
 	# that depend on a fixed address for common functions.
+	if echo $DEB_BUILD_ARCH | grep -q -v '^mips'; then
 	AC_MSG_CHECKING(if GCC supports -pie)
 	OLDCFLAGS="$CFLAGS"
 	CFLAGS="$CFLAGS -pie -fPIE"
@@ -157,6 +158,7 @@
 		AC_MSG_RESULT(yes),
 		AC_MSG_RESULT(no))
 	CFLAGS="$OLDCFLAGS"
+	fi
 
 	if test "x$with_optim" = x; then
 		# Add useful warning options for tracking down problems...
